tests/wasmfs/wasmfs_open.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+// FIXME: Merge with other existing close and open tests.
+
+int main() {
+  // Test writing to a file with a trailing slash.
+  int fd = open("/dev/stdout/", O_RDONLY);
```

Comment:
I just tried this out locally and it looks like this should not succeed (although it would if `/dev/stdout` were a directory).

Reply:
Works for Mac, but it appears that we just trim all slashes in the existing FS:

https://github.com/emscripten-core/emscripten/blob/d0bccda0e1305f4cfcf8f81c1d1eb568a57d0a76/src/library_fs.js#L138-L141